x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.test.ts

```diff
@@ -93,5 +95,48 @@ describe('ML - data recognizer', () => {
         },
       ]);
     });
+
+    it('should filter configs', () => {
+      const configs = [
+        {
+          module: { tags: ['security'] },
+        },
+        {
+          module: { tags: ['security', 'observability'] },
+        },
+        {
+          module: { tags: ['security', 'logs'] },
+        },
+        {
+          module: { tags: ['search'] },
+        },
+        {
+          module: { tags: [] },
+        },
+        {
+          module: { tags: [] },
+        },
+        {
+          module: {},
+        },
+      ] as unknown as Config[];
+
+      const c1 = filterConfigs(configs, null, []);
+      expect(c1.length).toBe(7);
```

Comment:
Would be good to add comments what we are testing for. Also, recommendation to add explicit assertions of the content of the expected results, not just the length for clarity.

Reply:
I'm reluctant to match against the full response mainly because we'd have to keep the the various JSON responses on disk to compare against and they can be quite large. For example the full list response is 8k line long.

Reply:
Updated in https://github.com/elastic/kibana/pull/168721/commits/3f57d2408bdcfe73a251bdd0a13b6e1fc17818ad